src/backend/mod.rs

```diff
@@ -16,15 +16,25 @@ pub use self::crossterm::CrosstermBackend;
 mod test;
 pub use self::test::TestBackend;
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum ClearType {
+    All,
+    AfterCursor,
+    BeforeCursor,
+    CurrentLine,
+    UntilNewLine,
+}
+
 pub trait Backend {
     fn draw<'a, I>(&mut self, content: I) -> Result<(), io::Error>
     where
         I: Iterator<Item = (u16, u16, &'a Cell)>;
+    fn append_lines(&mut self, n: u16) -> io::Result<()>;
     fn hide_cursor(&mut self) -> Result<(), io::Error>;
     fn show_cursor(&mut self) -> Result<(), io::Error>;
     fn get_cursor(&mut self) -> Result<(u16, u16), io::Error>;
     fn set_cursor(&mut self, x: u16, y: u16) -> Result<(), io::Error>;
-    fn clear(&mut self) -> Result<(), io::Error>;
+    fn clear(&mut self, clear_type: ClearType) -> Result<(), io::Error>;
     fn size(&self) -> Result<Rect, io::Error>;
     fn flush(&mut self) -> Result<(), io::Error>;
 }
```

Comment:
Assuming we don't want to break this trait, we could do
```diff
     fn draw<'a, I>(&mut self, content: I) -> Result<(), io::Error>
     where
         I: Iterator<Item = (u16, u16, &'a Cell)>;
-    fn append_lines(&mut self, n: u16) -> io::Result<()>;
+    fn append_lines(&mut self, _n: u16) -> io::Result<()> {
+        Ok(())
+    }
     fn hide_cursor(&mut self) -> Result<(), io::Error>;
     fn show_cursor(&mut self) -> Result<(), io::Error>;
     fn get_cursor(&mut self) -> Result<(u16, u16), io::Error>;
     fn set_cursor(&mut self, x: u16, y: u16) -> Result<(), io::Error>;
-    fn clear(&mut self, clear_type: ClearType) -> Result<(), io::Error>;
+    fn clear(&mut self) -> Result<(), io::Error>;
+    fn clear_to(&mut self, clear_type: ClearType) -> Result<(), io::Error> {
+        match clear_type {
+            ClearType::All => self.clear(),
+            ClearType::AfterCursor
+            | ClearType::BeforeCursor
+            | ClearType::CurrentLine
+            | ClearType::UntilNewLine => Err(io::Error::new(
+                io::ErrorKind::Other,
+                format!("clear_type [{clear_type:?}] not supported with this backend"),
+            )),
+        }
+    }
     fn size(&self) -> Result<Rect, io::Error>;
     fn flush(&mut self) -> Result<(), io::Error>;
 }
```

But I'm not sure if that would add confusion

Reply:
Yeah I definitely don't think we want to break the trait.
But I don't think ```clear_to``` is the best name for the new function although to be fair I can't come up with anything better.
You could either leave it as is or if you find a better name change it. Maybe add a doc line that explains that ```clear``` will ClearType::All where as  ```clear_to``` can be specified.

Maybe ```clear_specified``` ?

Reply:
`clear_to` with documentation would work, though I also think `clear_area` or `clear_region` might also make sense?

Reply:
`clear_region` implemented in #119! :)

Reply:
clear_region works for me too! I'll update my PR tomorrow. I'll also add some documentation. Won't hurt